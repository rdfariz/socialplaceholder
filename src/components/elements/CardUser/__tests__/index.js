import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Card from '../';
import { shallow } from 'enzyme';

describe('src/components/elements/CardUser', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(
      <Card title="test" id="1" body="testbody" isDetail />
    );
    expect(tree).toMatchSnapshot();
  });

  test('should return handleClick when card clicked', () => {
    const wrapper = shallow(<Card title="test" id="1" body="testbody" />);
    const card = wrapper.find('div');
    if (card.exists()) {
      card.first().simulate('click');
    }
  });
});
