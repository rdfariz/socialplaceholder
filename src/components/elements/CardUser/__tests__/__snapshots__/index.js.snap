// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/CardUser render 1`] = `
<div
  className=" flex"
>
  <div
    className="card_user w-full p-5 rounded-lg border bg-gradient-to-r from-pink-500 via-red-500 to-yellow-400 ..."
    onClick={[Function]}
  >
    <div
      className="text-white"
    >
      <h2
        className="text-lg font-bold"
      >
        test
      </h2>
      <p
        className="text-sm"
      >
        testbody
      </p>
    </div>
  </div>
</div>
`;
